nomad/structs/structs.go

```diff
@@ -1527,6 +1527,10 @@ func (s *Service) Validate() error {
 	}
 
 	for _, c := range s.Checks {
+		if s.PortLabel == "" && (c.Type == ServiceCheckTCP || c.Type == ServiceCheckHTTP) {
```

Comment:
I wonder if it would be better to pass down the PortLabel information into the check? Or to add a method to the check which returns whether it requires a port. The inspection of the type seems hacky.